[reorg] new Recipes section; Basics streamlined (#2550)
withastro/docs: src/content/docs/en/core-concepts/astro-components.mdx

i18nReady: true
---

> **Astro components** are the basic building blocks of any Astro project. They are HTML-only templating components with no client-side runtime. You can spot an Astro component by its file extension: `.astro`.

Reviewer: I know this wasn't introduced in this PR, but this sentence is misleading, as:
- "HTML-only" might sound like we're excluding CSS and JS, which can both be included in an Astro component
- You can include script tags for a client-side runtime

First thought:
```suggestion
**Astro components** are the basic building blocks of any Astro project. They are templating components that are built ahead of time, rather than on the client. You can spot an Astro component by its file extension: `.astro`.
```

## Component Structure

> An Astro component is made up of two main parts: the **Component Script** and the **Component Template**. Each part performs a different job, but together they provide a framework that is both easy to use and expressive enough to handle whatever you might want to build.

Reviewer: Now is a good time to standardize to **frontmatter (script)**, which is used in the tutorial and more prevalent in our community.

<!-- Component Template (HTML + JS Expressions) -->
```

### The Component Script

> Astro uses a code fence (`---`) to identify the component script in your Astro component. If you've ever written Markdown before, you may already be familiar with a similar concept called *frontmatter.* Astro's idea of a component script was directly inspired by this concept.

Reviewer: Whether or not we change the name, I don't think the Markdown frontmatter comparison is helpful enough when they've heard of it to justify the cases when they haven't. They're more visually similar than conceptually similar.

If you write plain HTML here, your component will render that HTML in any Astro page it is imported and used.

> However, [Astro's component template syntax](/en/core-concepts/astro-syntax/) also supports **JavaScript expressions**, **imported components** and [**special Astro directives**](/en/reference/directives-reference/). Data and values defined (during rendering) in the component script can be used in the component template to produce dynamically-created HTML.

Author: just checking, the only thing the original provides, the one that says at build time, is the reinforcement that while these are dynamic values they're not updating ones. I'm happy to leave this out if we are comfortable losing that one extra nudge that these values are read and injected into the HTML template only once.

> However, [Astro's component template syntax](/en/core-concepts/astro-syntax/) also supports **JavaScript expressions**, **imported components** and [**special Astro directives**](/en/reference/directives-reference/). Data and values defined (during rendering) in the component script can be used in the component template to produce dynamically-created HTML.

Reviewer: Yeah, I don't think the original provides that in an accurate/elegant way, so I think we leave it out

<p class:list={["add", "dynamic", {classNames: true}]} />
```

> ## JSX-like Expressions

Reviewer: Totally agree with moving this into its own guide!
